src/main/kotlin/no/nav/syfo/syfosmvarsel/varselutsending/pdl/service/PdlPersonService.kt

```diff
@@ -0,0 +1,35 @@
+package no.nav.syfo.syfosmvarsel.varselutsending.pdl.service
+
+import no.nav.syfo.client.StsOidcClient
+import no.nav.syfo.syfosmvarsel.varselutsending.pdl.client.PdlClient
+import no.nav.syfo.syfosmvarsel.varselutsending.pdl.client.model.Gradering
+import no.nav.syfo.syfosmvarsel.varselutsending.pdl.error.PersonNotFoundInPdl
+import org.slf4j.LoggerFactory
+
+class PdlPersonService(private val pdlClient: PdlClient, private val stsOidcClient: StsOidcClient) {
+    companion object {
+        private val log = LoggerFactory.getLogger(PdlPersonService::class.java)
+    }
+    suspend fun harDiskresjonskode(fnr: String, sykmeldingsId: String): Boolean {
+        val stsToken = stsOidcClient.oidcToken().access_token
+        val pdlResponse = pdlClient.getPerson(fnr, stsToken)
+
+        if (pdlResponse.errors != null) {
+            pdlResponse.errors.forEach {
+                log.error("PDL kastet error: {} ", it)
```

Comment:
Hva slags feil kan dette være? Hvis det er feil som gjør at vi kan ha hentet uriktig informasjon, eller tekniske feil, må vi kaste en exception her sånn at vi ikke går videre :)

Reply:
Ja, her vil det typisk være at PDL ikke finner brukeren. F.eks. vil den returnere dette 

`{
  "errors": [
    {
      "message": "Fant ikke person",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "hentIdenter"
      ],
      "extensions": {
        "code": "not_found",
        "classification": "ExecutionAborted"
      }
    },
    {
      "message": "Fant ikke person",
      "locations": [
        {
          "line": 9,
          "column": 3
        }
      ],
      "path": [
        "hentPerson"
      ],
      "extensions": {
        "code": "not_found",
        "classification": "ExecutionAborted"
      }
    }
  ],
  "data": {
    "hentIdenter": null,
    "hentPerson": null
  }
}`

hvis du spør etter en bruker den ikke finner.

Å kaste en Exception her er nok smart.

Reply:
Hvis bruker ikke finnes i PDL oppdager vi jo det i neste if-sjekk..? Men hvis det kan være andre feil derimot er jo denne nyttig da, men i så fall bør vi kanskje flytte denne sjekken slik at vi først sjekker om hentPerson er null, hvis ikke vil den aldri bli kjørt :) 

Og så trenger vi kanskje ikke hente noe mer enn message for error, hvis det er mulig å unngå..? :)

Reply:
Ja, det kan være alle mulige feil PDL returnerer. Jeg har ikke oversikt over alle mulige feil, så jeg syns det er fint å logge ev. feil som oppstår, så blir det forhåpentligvis litt lettere å debugge.

Hvis vi får feil så kan vi anta at vi ikke har klart å hente ut person her, for det vil typisk være at vi ikke fant personen, at brukeren vår ikke er autorisert til å se dataene, eller at PDL ikke klarer å autentisere oss. Jeg tror ikke vi vil få både erros og data i samme resultat for denne spørringen.

Reply:
At vi ikke finner person er et spesialtilfelle av at PDL returnerte feil, så sånn sett bør den sjekken gjøres først. 

Hvis det er noen som helst tvil om kvaliteten på det vi har hentet fra PDL MÅ det kastes feil sånn at vi avbryter og prøver på nytt. Akkurat diskresjonskode har ekstremt alvorlige konsekvenser hvis vi gjør noe feil, i aller verste fall kan folk bli drept, så her må vi være 100% sikre på at vi gjør det riktig.

Reply:
Andreas har forresten ikke tatt med hentingen av feilkoder da han satte opp dette første gangen i egenmeldt-backend, kanskje han gjorde noen undersøkelser som avklarte at det ikke var nødvendig?

Reply:
Nei, det er ikke nødvendig. Men er det ikke kjekt å kunne se hva PDL sier hvis noe feiler pga. PDL?

Reply:
Jeg har fjernet uthenting av feilmeldinger fra PDL.